lib/saml/post_url_service.rb

```diff
@@ -39,7 +39,7 @@ def initialize(saml_settings, session: nil, user: nil, params: {}, loa3_context:
     end
 
     def should_uplevel?
-      user.loa[:current] < user.loa[:highest]
+      user.loa[:current] < user.loa[:highest] || (!user.verified_at&.empty? && user.loa[:current] < LOA::THREE)
```

Comment:
I wonder if we should refactor this just a bit to make it more clear what's happening here, maybe something like:
`user.verified_at.present? ? user.loa[:current] < LOA::THREE : user.loa[:current] < user.loa[:highest]`